src/ASM/LR/ASMu2Drecovery.C

```diff
@@ -153,23 +153,24 @@ bool ASMu2D::assembleL2matrices (SparseMatrix& A, StdVector& B,
       int ielp = group[t][e];
       const LR::Element* elm = proj->getElement(ielp);
       int iel = lrspline->getElementContaining(elm->midpoint())+1;
+      int ielG = geo->getElementContaining(elm->midpoint())+1;
 
       if (continuous)
       {
         // Set up control point (nodal) coordinates for current element
         if (!this->getElementCoordinates(Xnod,iel)) {
           ok = false;
           continue;
-        } else if ((dA = 0.25*this->getParametricArea(iel)) < 0.0) {
+        } else if ((dA = 0.25*elm->area()) < 0.0) {
```

Comment:
Consider same change on other uses of `getParametricArea()` for the LR-classes. Maybe we don't need that method at all since we probably always fetches the elm pointer for other uses anyway.

Reply:
Yeah, done.

Reply:
great, the same sing for ASMu3D also then, maybe, although there the `getParametricVolume()` method in addition checks if `MNPC[iel-1].empty())`. Not sure if this is needed, or why the same was not done for 2D. At least there is an inconsitency here that should be sorted out, one way or the other.

Reply:
i don't think that makes sense at all. We don't have empty knotspans represented as elements with LR so we also do not have empty mnpc-entries. i think it's just a copy-pasta from the struct code.

Reply:
the fact that the method is completely unused also kinda shortcuts the discussion :)

Reply:
yes, let me rephrase. it's completely unused after my refactorings.